http/server.go

```diff
@@ -254,7 +254,9 @@ func (h *handler) PublicRand(w http.ResponseWriter, r *http.Request) {
 	roundExpectedTime = time.Unix(chain.TimeOfRound(info.Period, info.GenesisTime, roundN), 0)
 
 	if roundExpectedTime.After(time.Now().Add(info.Period)) {
+		timeToExpected := int(roundExpectedTime.Sub(time.Now()).Seconds())
```

Comment:
That's correct but that's not what you are checking in the if condition above (which seems wrong). Why is that ?

Reply:
If it's the immediate next event (so in the next next 30 seconds) we attempt to request the upcoming round from the server, which should block until it's available. If that blocking fails, we return the error on line 272